File: JanperCab/5-JanperCab/ClientApp/src/app/_models/DuraformProcess/DuraformProcessCleaningDto.ts
import { MachineCleaningDto } from './../machine/MachineCleaningDto';
import { DuraformProcessDto } from './DuraformProcessDto';

export class DuraformProcessCleaningDto extends DuraformProcessDto {
  machineId: number;

  machineCleaning: MachineCleaningDto;

  getStatus(): string {
    let status = this.startTime && this.endTime ? 'Cleaned' : 'Cleaning';

    if (this.machineCleaning) {
      status += ` - ${this.machineCleaning.name}`;
    }

    return status;
  }
}

File: JanperCab/3-Application/Dtos/Process/DuraformProcessCleaningDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class DuraformProcessCleaningDto : DuraformProcessDto
    {
        public int? MachineId { get; set; }

        public MachineCleaningDto MachineCleaning { set; get; }
    }
}